langchain4j-mistral-ai/src/main/java/dev/langchain4j/model/mistralai/MistralAiChatModelName.java

```diff
@@ -12,23 +12,26 @@
  * The available chat completion models are:
  * </p>
  * <ul>
- *   <li>{@link #MISTRAL_TINY} - powered by Mistral-7B-v0.2</li>
- *   <li>{@link #MISTRAL_SMALL} - powered by Mixtral-8X7B-v0.1</li>
- *   <li>{@link #MISTRAL_MEDIUM} - currently relies on an internal prototype model</li>
+ *   <li>{@link #OPEN_MISTRAL_7B} - aka mistral-tiny-2312</li>
+ *   <li>{@link #OPEN_MIXTRAL_8x7B} - aka mistral-small-2312</li>
+ *   <li>{@link #MISTRAL_SMALL_LATEST} - aka mistral-small-2402</li>
+ *   <li>{@link #MISTRAL_MEDIUM_LATEST} - aka mistral-medium-2312</li>
+ *   <li>{@link #MISTRAL_LARGE_LATEST} - aka mistral-large-2402</li>
  * </ul>
  *
- * @see <a href="https://docs.mistral.ai/platform/endpoints/">Mistral AI Endpoints</a>
+ * @see <a href="https://docs.mistral.ai/guides/model-selection/">Mistral Model Selection</a>
  */
 public enum MistralAiChatModelName {
 
-    // powered by Mistral-7B-v0.2
+    OPEN_MISTRAL_7B("open-mistral-7b"), // aka mistral-tiny-2312
+    @Deprecated
     MISTRAL_TINY("mistral-tiny"),
-
-    // powered Mixtral-8X7B-v0.1
-    MISTRAL_SMALL("mistral-small"),
-
-    // currently relies on an internal prototype model
-    MISTRAL_MEDIUM("mistral-medium");
```

Comment:
`MISTRAL_MEDIUM` can still be used, can you keep it and mark deprecated? This can break for someone.

Reply:
yeah, you're right. I deleted it because `MISTRAL_MEDIUM` It is no longer online, in fact it returns an error when you call it from the api. But you are right, it is better for the user to capture this exception than to delete it.

Reply:
I tested it yesterday and `mistral-medium` works fine. Just tested it again and it works:

```
DEBUG: Request:
- method: POST
- url: https://api.mistral.ai/v1/chat/completions
- headers: [Authorization: Bearer go...sC]
- body: {
  "model": "mistral-medium",
  "messages": [
    {
      "role": "user",
      "content": "What is the capital of Peru?"
    }
  ],
  "temperature": 0.1,
  "stream": false
}
2024-03-20 09:09:09 [main] dev.langchain4j.model.mistralai.MistralAiResponseLoggingInterceptor.log()
DEBUG: Response:
- status code: 200
- headers: [date: Wed, 20 Mar 2024 08:09:09 GMT], [content-type: application/json], [ratelimitbysize-query-cost: 32008], [ratelimitbysize-limit: 2000000], [ratelimitbysize-remaining: 1967992], [x-ratelimitbysize-remaining-month: 9996748905], [ratelimitbysize-reset: 1], [x-ratelimitbysize-limit-minute: 2000000], [x-ratelimitbysize-remaining-minute: 1967992], [x-ratelimitbysize-limit-month: 10000000000], [access-control-allow-origin: *], [x-kong-upstream-latency: 10573], [x-kong-proxy-latency: 4], [x-kong-request-id: 9e76635dd2a9ef34b103554ddb47ee2b], [cf-cache-status: DYNAMIC], [server: cloudflare], [cf-ray: 867437a99bb01d8a-FRA], [alt-svc: h3=":443"; ma=86400]
- body: {"id":"6d1d6eca04d340a2b85eb2cf51ac2dac","object":"chat.completion","created":1710922139,"model":"mistral-medium","choices":[{"index":0,"message":{"role":"assistant","content":"The capital of Peru is Lima. It was founded by Spanish conquistador Francisco Pizarro in 1535 and is the country's largest city, as well as its economic, political, and cultural hub. Lima is known for its rich history, vibrant culture, and delicious cuisine, which is a fusion of indigenous, Spanish, African, and Asian influences. Some of the city's most popular tourist attractions include the historic center of Lima, which is a UNESCO World Heritage site, the Larco Museum, which houses an impressive collection of pre-Columbian artifacts, and the Miraflores district, which is known for its stunning ocean views, upscale shopping, and lively nightlife."},"finish_reason":"stop","logprobs":null}],"usage":{"prompt_tokens":15,"total_tokens":176,"completion_tokens":161}}
```